test/libsyncengine/update_detection/file_system_observer/testlocalfilesystemobserverworker.cpp

```diff
@@ -452,16 +466,73 @@ void TestLocalFileSystemObserverWorker::testLFSOFastMoveDeleteMove() { // MS Off
     CPPUNIT_ASSERT(_syncPal->snapshot(ReplicaSide::Local)->exists(std::to_string(fileStat.inode)));
 }
 
-bool MockLocalFileSystemObserverWorker::waitForUpdate(int64_t timeoutMs) const {
+void TestLocalFileSystemObserverWorker::testLFSOFastMoveDeleteMoveWithEncodingChange() {
+    using namespace testhelpers;
+
+    LOGW_DEBUG(_logger, L"***** Test fast move/delete with enconding change*****"); // Behaviour of MS office apps on macOS
+    _syncPal->_localFSObserverWorker->stop();
+    _syncPal->_localFSObserverWorker->waitForExit();
```

Comment:
Added because there is a crash otherwise. Seemingly related to the fact that the observer thread didn't terminate at the time its destructor is called.

Reply:
Ok